plugin/trino-hive/src/test/java/io/trino/plugin/hive/BaseTestHiveInsertOverwrite.java

```diff
@@ -0,0 +1,216 @@
+        copyTpchNationToTable(testTable);
+        assertOverwritePartition(externalTableName);
+    }
+
+    protected void assertInsertFailsOnUnsupportedWritingToTarget(String testTable, String expectedMessageRegExp)
+    {
+        assertQueryFails(
+                format("INSERT INTO %s " +
+                                "SELECT name, comment, nationkey, regionkey " +
+                                "FROM tpch.tiny.nation",
+                        testTable),
+                expectedMessageRegExp);
+        computeActual(format("DROP TABLE %s", testTable));
+    }
+
+    protected void assertOverwritePartition(String testTable)
+    {
+        computeActual(format(
+                "INSERT INTO %s VALUES " +
+                        "('POLAND', 'Test Data', 25, 5), " +
+                        "('CZECH', 'Test Data', 26, 5)",
+                testTable));
+        MaterializedResult materializedRows = computeActual(format("SELECT count(*) FROM %s WHERE regionkey=5", testTable));
+        assertThat(materializedRows)
+                .isEqualTo(resultBuilder(getSession(), materializedRows.getTypes())
+                        .row(2L)
+                        .build());
+
+        computeActual(format("INSERT INTO %s values('POLAND', 'Overwrite', 25, 5)", testTable));
+        materializedRows = computeActual(format("SELECT count(*) FROM %s WHERE regionkey=5", testTable));
+        assertThat(materializedRows)
+                .isEqualTo(resultBuilder(getSession(), materializedRows.getTypes())
+                        .row(1L)
+                        .build());
+        computeActual(format("DROP TABLE %s", testTable));
+    }
+
+    protected String getTestTableName()
+    {
+        return format("hive.%s.%s", HIVE_TEST_SCHEMA, "nation_" + randomTableSuffix());
+    }
+
+    protected String getCreateTableStatement(String tableName, String... propertiesEntries)
+    {
+        return getCreateTableStatement(tableName, Arrays.asList(propertiesEntries));
+    }
+
+    protected String getCreateTableStatement(String tableName, List<String> propertiesEntries)
+    {
+        return format(
+                "CREATE TABLE %s (" +
+                        "    name varchar(25), " +
+                        "    comment varchar(152),  " +
+                        "    nationkey bigint, " +
+                        "    regionkey bigint) " +
+                        (propertiesEntries.size() < 1 ? "" : propertiesEntries
+                                .stream()
+                                .collect(joining(",", "WITH (", ")"))),
+                tableName);
+    }
+
+    protected void copyTpchNationToTable(String testTable)
+    {
+        computeActual(
+                format("INSERT INTO %s " +
+                                "SELECT name, comment, nationkey, regionkey " +
+                                "FROM tpch.tiny.nation",
+                        testTable));
+    }
```

Comment:
nit: 
```
        computeActual("INSERT INTO " + testTable + " SELECT * FROM tpch.tiny.nation");
```

Reply:
Order of columns is changed so I can't use * explicit. But inlined testTable name.